src/renderer/services/plugin-manager.js

```diff
@@ -15,40 +24,81 @@ if (process.env.NODE_ENV === 'production') {
 
 export class PluginManager {
   constructor () {
+    this.app = null
+    this.adapter = null
+    this.pluginsPath = null
     this.plugins = {}
     this.hasInit = false
     this.vue = null
   }
 
+  setAdapter (adapter) {
+    this.adapter = adapters[upperFirst(adapter) + 'Adapter']
+  }
+
+  setApp (app) {
+    this.app = app
+  }
+
   setVue (vue) {
     this.vue = vue
   }
 
   async init (app) {
-    this.app = app
+    this.setApp(app)
+
+    this.pluginsPath = process.env.NODE_ENV !== 'development' ? PLUGINS.path : PLUGINS.devPath
 
-    await this.app.$store.dispatch('plugin/init')
-    await this.fetchPluginsFromPath(
-      process.env.NODE_ENV !== 'development' ? PLUGINS.path : PLUGINS.devPath
-    )
+    await this.app.$store.dispatch('plugin/reset')
+    await this.fetchPlugins()
 
     this.hasInit = true
 
     await this.app.$store.dispatch('plugin/loadPluginsForProfiles')
   }
 
+  async deletePlugin (pluginId) {
+    if (!this.hasInit) {
+      throw new errors.NotInitiatedError()
+    }
+
+    const plugin = this.plugins[pluginId]
+    if (!plugin) {
+      throw new errors.PluginNotFoundError(pluginId)
+    }
+
+    const parentDir = path.dirname(plugin.fullPath)
+    const pluginCount = fs.readdirSync(parentDir).filter(entry => {
+      return entry !== '.DS_Store'
```

Comment:
Will need to exclude `.cache` too. Perhaps exclude anything starting with `.`

Reply:
The `.cache`  folder won't matter in this case since we check `parentDir !== this.pluginsPath` anyway, so the root plugin directory is never deleted. But there might be other temporary files, so I'll exclude anything starting with a dot as proposed.